Source/WTF/wtf/StdLibExtras.h

```diff
@@ -1574,6 +1574,39 @@ template<typename T, typename U> constexpr auto forward_like(U&& value) -> detai
 template<typename T, typename U> constexpr auto forward_like_preserving_const(U&& value) -> detail::forward_like_preserving_const_impl<T, U> { return static_cast<detail::forward_like_preserving_const_impl<T, U>>(value); }
 } // namespace WTF
 
+#if defined(__GLIBCXX__) && !defined(__cpp_lib_ranges_zip)
+#include <wtf/ZippedRange.h>
+
+namespace std::ranges {
+
+struct _Zip {
+    template<typename... _Ts>
+    requires (sizeof...(_Ts) <= 2)
+    constexpr auto
+    operator() [[nodiscard]] (_Ts&&... __ts) const {
+        if constexpr (!sizeof...(_Ts)) {
+            return views::empty<tuple<>>;
+        } else if constexpr (sizeof...(_Ts) == 1) {
+            return [](auto&& arg1, auto&&...) {
+                return std::forward<decltype(arg1)>(arg1);
+            }(std::forward<_Ts>(__ts)...);
+        } else {
+            return [](auto&& arg1, auto&& arg2, auto&&...) {
+                return zippedRange(std::forward<decltype(arg1)>(arg1), std::forward<decltype(arg2)>(arg2));
+            }(std::forward<_Ts>(__ts)...);
+        }
+    }
+};
+
+namespace views {
+    inline constexpr _Zip zip;
+}
+
+using WTF::zippedRange;
+
+} // namespace std::ranges
+#endif // defined(__GLIBCXX__) && !defined(__cpp_lib_ranges_zip)
```

Comment:
This very cool, but I since WTF::ZippedRange only supports two operands, I think this is likely to cause confusion.

I think either we should either:

a) pull in a complete std::views::zip implementation
b) replace the one use directly with WTF::ZippedRange (probably the easiest)
c) update WTF::ZippedRange to support an arbitrary number of ranges